### PR TITLE
8334165: Remove serialVersionUID compatibility logic from JMX

### PR DESCRIPTION
**Pull request by kevinjwalls**
Remove very very old serialization compatibility logic from JMX.

This relates to keeping the JDK's JMX implementation compatible with JMX versions from when JMX was a separate component, before being integrated into JDK 5.  It should all be removed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8339559](https://bugs.openjdk.org/browse/JDK-8339559) to be approved

### Issues
 * [JDK-8334165](https://bugs.openjdk.org/browse/JDK-8334165): Remove serialVersionUID compatibility logic from JMX (**Enhancement** - P4)
 * [JDK-8339559](https://bugs.openjdk.org/browse/JDK-8339559): Remove serialVersionUID compatibility logic from JMX (**CSR**)

### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20856/head:pull/20856` \
`$ git checkout pull/20856`

Update a local copy of the PR: \
`$ git checkout pull/20856` \
`$ git pull https://git.openjdk.org/jdk.git pull/20856/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20856`

View PR using the GUI difftool: \
`$ git pr show -t 20856`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20856.diff">https://git.openjdk.org/jdk/pull/20856.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20856#issuecomment-2331672762)

### CODE REVIEW

**Review by dfuch — COMMENTED** [review]
Wow. I hadn't realized so many classes were impacted. I thought there was just a few!

Look good in general. It could be good to write a test that sets the property with 1.0 and 1.1, and that verifies that something (e.g. an ObjectName?) serialized with property=1.0 can be deserialized with property=1.1, and conversely.

That would validate that setting the property no longer has any effect.

**Comment by kevinjwalls** [comment]
Thanks Daniel @dfuch , we can use a version of what we had before, there was a simple check on ObjectName fields.  I added it, will fail in a JDK that respects jmx.serial.version.

**Review by dfuch — APPROVED** [review]
LGTM  - might be good to have an other pait of eyes looking at it before integrating...

File: src/java.management/share/classes/javax/management/modelmbean/InvalidTargetObjectTypeException.java [hunk 1]
```diff
@@ -46,61 +46,13 @@
...
-    private static final ObjectStreamField[] oldSerialPersistentFields =
-    {
-      new ObjectStreamField("msgStr", String.class),
-      new ObjectStreamField("relatedExcept", Exception.class)
-    };
-    //
-    // Serializable fields in new serial form
-    private static final ObjectStreamField[] newSerialPersistentFields =
+    private static final long serialVersionUID = 1190536278266811217L;
+    private static final ObjectStreamField[] serialPersistentFields =
```
Comment (dfuch) [inline src/java.management/share/classes/javax/management/modelmbean/InvalidTargetObjectTypeException.java:52]: The API doc comment for the serialized fields dhould be preserved.
```suggestion
    /**
     * @serialField exception Exception Encapsulated {@link Exception}
     */
    private static final ObjectStreamField[] serialPersistentFields =
```

File: src/java.management/share/classes/javax/management/modelmbean/ModelMBeanAttributeInfo.java [hunk 2]
```diff
@@ -116,63 +116,15 @@
...
-            serialVersionUID = oldSerialVersionUID;
-        } else {
-            serialPersistentFields = newSerialPersistentFields;
-            serialVersionUID = newSerialVersionUID;
-        }
-    }
-    //
-    // END Serialization compatibility stuff
+        private static final long serialVersionUID = 6181543027787327345L;
+        private static final ObjectStreamField[] serialPersistentFields =
```
Comment (dfuch) [inline src/java.management/share/classes/javax/management/modelmbean/ModelMBeanAttributeInfo.java:124]: API doc coment for serial persistent fields should have been preserved.
```suggestion
        /**
         * @serialField attrDescriptor Descriptor The {@link Descriptor}
         * containing the metadata corresponding to this attribute
         */
        private static final ObjectStreamField[] serialPersistentFields =
```

**Review by dfuch — CHANGES_REQUESTED** [review]

File: src/java.management/share/classes/javax/management/modelmbean/ModelMBeanNotificationInfo.java [hunk 3]
```diff
@@ -96,64 +96,15 @@
...
-    private static final ObjectStreamField[] oldSerialPersistentFields =
-    {
-      new ObjectStreamField("notificationDescriptor", Descriptor.class),
-      new ObjectStreamField("currClass", String.class)
-    };
-    //
-    // Serializable fields in new serial form
-    private static final ObjectStreamField[] newSerialPersistentFields =
+    private static final long serialVersionUID = -7445681389570207141L;
+    private static final ObjectStreamField[] serialPersistentFields =
```
Comment (dfuch) [inline src/java.management/share/classes/javax/management/modelmbean/ModelMBeanNotificationInfo.java:104]: Same here:
```suggestion
    /**
     * @serialField notificationDescriptor Descriptor The descriptor
     *   containing the appropriate metadata for this instance
     */
    private static final ObjectStreamField[] serialPersistentFields =
```

**Review by dfuch — CHANGES_REQUESTED** [review]

**Comment by kevinjwalls** [comment]
thanks yes, updated 4 missing serialField comments

**Review by dfuch — COMMENTED** [review]
OK - thanks. LGTM now.

**Review by dfuch — APPROVED** [review]

**Pull request closed without merging**